src/allmydata/scripts/tahoe_invite.py

```diff
@@ -0,0 +1,101 @@
+import json
+from os.path import join
+from ConfigParser import NoSectionError
+
+from twisted.python import usage
+from twisted.internet import defer, reactor
+
+from wormhole import wormhole
+
+from allmydata.util import configutil
+
+
+INVITE_SEPARATOR = "+"
+
+class InviteOptions(usage.Options):
+    synopsis = "[options] <nickname>"
+    description = "Create a client-only Tahoe-LAFS node (no storage server)."
+
+    optParameters = [
+        ("needed", None, None, "How many shares are needed to reconstruct files from this node"),
+        ("happy", None, None, "Distinct storage servers new node will upload shares to"),
+        ("total", None, None, "Total number of shares new node will upload"),
+    ]
+
+    def parseArgs(self, *args):
+        if len(args) != 1:
+            raise usage.UsageError(
+                "Provide a single argument: the new node's nickname"
+            )
+        self['nick'] = args[0].strip()
+
+
+@defer.inlineCallbacks
+def invite(options):
+    basedir = unicode(options.parent['node-directory'])
+    config = configutil.get_config(join(basedir, 'tahoe.cfg'))
+    out = options.stdout
+    err = options.stderr
+
+    try:
+        introducer_furl = config.get('client', 'introducer.furl')
+    except NoSectionError:
+        # we're not a client; maybe this is running *on* the introducer?
+        try:
+            with open(join(basedir, "private", "introducer.furl"), "r") as f:
+                introducer_furl = f.read().strip()
+        except IOError:
+            print >>err, "Can't find introducer fURL in tahoe.cfg nor " + \
+                "{}/private/introducer.furl".format(basedir)
+            raise SystemExit(1)
+
+    nick = options['nick']
+
+    config = {
+        "needed": options["needed"] or config.get('client', 'shares.needed'),
+        "total": options["total"] or config.get('client', 'shares.total'),
+        "happy": options["happy"] or config.get('client', 'shares.happy'),
+        "nickname": nick,
+        "introducer": introducer_furl,
+    }
+
+    wh = wormhole.create(
+        appid=u"tahoe-lafs.org/lafs",
+        relay_url=u"ws://wormhole.tahoe-lafs.org:4000/v1",
```

Comment:
yeah :) I now made it a global option. maybe overkill? but might be useful when testing etc...